### DISCUSSION TITLE
Is it possible to have a cloup.argument in an option_group?

### DISCUSSION THREAD

**Discussion by wabiloo** [janluke/cloup#170] [Q&A] [answered]
In click, I have the following:
```
def search_options(fn):
    @click.argument("single_term", required=False)
    @click.option(
        "-t",
        "--term",
        "search_terms",
        multiple=True,
        required=False,
        default=(),
        help="Search term",
    )
    @click.option(
        "-i",
        "--in",
        "search_fields",
        multiple=True,
        required=False,
        default=(None,),
        help="Field in which to search for the term",
    )
    @functools.wraps(fn)
    def wrapper(*args, **kwargs):
        return fn(*args, **kwargs)

    return wrapper
```

This allows me to express the command line as:
`app search BLA`
or 
`app search -t BLA -i field1`

I would like to turn this into an option group (because there are other options in other parts of the code), but the first element is an argument, not an option.
Can I add it to an option group?

**Comment by janluke** [comment, marked as answer]
No, option groups are for flag options only. All positional arguments goes (optionally) in the "Positional arguments" section (if you specify `help` for at least one of them). You can still define constraints on all parameters with `@constraint`.

What happens in this case?
```
app search -t term_1 -i field_1 -t term_2 -t term_3 -i field_3
```
Is term_2 searched in `field_3`?

**Reply by wabiloo** [reply]
If there is a mismatch between the number of -t and -i options, I throw an error

**Reply by janluke** [reply]
In that case, a tuple argument would probably serve you better: `-t TERM FIELD`.
```python
terms, fields = zip(*term_options)  # unzip
```

Regarding your specific question, I'm sorry, what you want is not possible in Cloup and I'm fine with that because I think positional arguments should stay in their own help section (or should be described in the command help). But I must say that I also tend to avoid optional positional arguments.
